auth/api/grpc/endpoint_test.go

```
@@ -349,29 +478,543 @@ func TestDeletePolicy(t *testing.T) {
+	authAddr := fmt.Sprintf("localhost:%d", port)
+	conn, _ := grpc.Dial(authAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	client := grpcapi.NewClient(conn, time.Second)
+	cases := []struct {
+		desc  string
+		token string
+		pr    *magistrala.CountSubjectsReq
+		ar    *magistrala.CountSubjectsRes
+		err   error
+		code  codes.Code
+	}{
+		{
+			desc:  "count subjects with valid token",
+			token: validToken,
+			pr: &magistrala.CountSubjectsReq{
+				Domain:      domainID,
+				SubjectType: usersType,
+				Relation:    memberRelation,
+				Permission:  adminpermission,
+			},
+			ar: &magistrala.CountSubjectsRes{
+				Count: 1,
+			},
+			code: codes.OK,
+			err:  nil,
+		},
+		{
+			desc:  "count subjects with invalid token",
+			token: inValidToken,
+			pr: &magistrala.CountSubjectsReq{
+				Domain:      domainID,
+				SubjectType: usersType,
+				Relation:    memberRelation,
+				Permission:  adminpermission,
+			},
+			ar:   &magistrala.CountSubjectsRes{},
+			err:  svcerr.ErrAuthentication,
+			code: codes.Unauthenticated,
+		},
+	}
+	for _, tc := range cases {
+		repoCall := svc.On("CountSubjects", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(int(tc.ar.Count), tc.err)
+		apr, err := client.CountSubjects(context.Background(), tc.pr)
+		if apr != nil {
+			assert.Equal(t, tc.ar, apr, fmt.Sprintf("%s: expected %v got %v", tc.desc, tc.ar, apr))
+		}
+		e, ok := status.FromError(err)
 		assert.True(t, ok, "gRPC status can't be extracted from the error")
 		assert.Equal(t, tc.code, e.Code(), fmt.Sprintf("%s: expected %s got %s", tc.desc, tc.code, e.Code()))
-		assert.Equal(t, tc.dpr.GetDeleted(), dpr.GetDeleted(), fmt.Sprintf("%s: expected %v got %v", tc.desc, tc.dpr.GetDeleted(), dpr.GetDeleted()))
+		repoCall.Unset()
+	}
+}
+
+func TestListPermissions(t *testing.T) {
+	authAddr := fmt.Sprintf("localhost:%d", port)
+	conn, _ := grpc.Dial(authAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Don't ignore this error handle it. Same as other cases